components/producers/checkov/main_test.go

```diff
@@ -0,0 +1,105 @@
+	"github.com/stretchr/testify/require"
+
+	v1 "github.com/ocurity/dracon/api/proto/v1"
+	"github.com/ocurity/dracon/components/producers"
+)
+
+const (
+	sarifInputPath     = "exampleData/results_sarif.sarif"
+	cyclonedxInputPath = "exampleData/results_cyclonedx.json"
+)
+
+func TestRunSarif(t *testing.T) {
+	workspace, err := os.MkdirTemp("", "dracon")
+	require.NoError(t, err)
+	defer os.RemoveAll(workspace)
+	producers.OutFile = filepath.Join(workspace, "out.pb")
+	input, err := os.ReadFile(sarifInputPath)
+	require.NoError(t, err)
+	require.NoError(t, run(input))
+
+	_, err = os.Stat(producers.OutFile)
+	require.NoError(t, err)
+
+	in, err := os.ReadFile(producers.OutFile)
+	require.NoError(t, err)
+	var wrote v1.LaunchToolResponse
+	err = proto.Unmarshal(in, &wrote)
+	require.NoError(t, err)
+	expectedIssues := []*v1.Issue{
+		{
+			Target:      "code/cfngoat/cfngoat.yaml:891-892",
+			Type:        "CKV_SECRET_6",
+			Title:       "Base64 High Entropy String",
+			Severity:    v1.Severity_SEVERITY_HIGH,
+			Description: "MatchedRule: {\"id\":\"CKV_SECRET_6\",\"name\":\"Base64 High Entropy String\",\"shortDescription\":{\"text\":\"Base64 High Entropy String\"},\"fullDescription\":{\"text\":\"Base64 High Entropy String\"},\"defaultConfiguration\":{\"level\":\"error\"},\"help\":{\"text\":\"Base64 High Entropy String\\nResource: c00f1a6e4b20aa64691d50781b810756d6254b8e\"}} \n Message: Base64 High Entropy String",
+		}, {
+			Target:      "code/cfngoat/.github/workflows/checkov.yaml:1-1",
+			Type:        "CKV2_GHA_1",
+			Title:       "Ensure top-level permissions are not set to write-all",
+			Severity:    v1.Severity_SEVERITY_HIGH,
+			Description: "MatchedRule: {\"id\":\"CKV2_GHA_1\",\"name\":\"Ensure top-level permissions are not set to write-all\",\"shortDescription\":{\"text\":\"Ensure top-level permissions are not set to write-all\"},\"fullDescription\":{\"text\":\"Ensure top-level permissions are not set to write-all\"},\"defaultConfiguration\":{\"level\":\"error\"},\"help\":{\"text\":\"Ensure top-level permissions are not set to write-all\\nResource: on(build)\"}} \n Message: Ensure top-level permissions are not set to write-all",
+		}, {
+			Target:      "code/cfngoat/.github/workflows/main.yaml:1-1",
+			Type:        "CKV2_GHA_1",
+			Title:       "Ensure top-level permissions are not set to write-all",
+			Severity:    v1.Severity_SEVERITY_HIGH,
+			Description: "MatchedRule: {\"id\":\"CKV2_GHA_1\",\"name\":\"Ensure top-level permissions are not set to write-all\",\"shortDescription\":{\"text\":\"Ensure top-level permissions are not set to write-all\"},\"fullDescription\":{\"text\":\"Ensure top-level permissions are not set to write-all\"},\"defaultConfiguration\":{\"level\":\"error\"},\"help\":{\"text\":\"Ensure top-level permissions are not set to write-all\\nResource: on(build)\"}} \n Message: Ensure top-level permissions are not set to write-all",
+		},
+	}
+
+	foundIssues := 0
+	for _, wroteIssue := range wrote.Issues {
```

Comment:
this looks like it could be simplified with a sort on the target in both slices and then use the EqualValues function

Reply:
protos have a ton of fields that are conflicting, I only want to test for equality on what matters